libbeat/scripts/cmd/global_fields/main.go

```diff
@@ -39,18 +40,30 @@ func main() {
 		os.Exit(1)
 	}
 
-	err := os.MkdirAll(filepath.Join(*beatPath, "_meta"), 0755)
+	esBeats, err := os.Open(*esBeatsPath)
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "Cannot create _meta dir for %s: %+v\n", name, err)
+		fmt.Fprintf(os.Stderr, "Error opening elastic/beats: %+v\n", err)
+		os.Exit(1)
+	}
+	beat, err := os.Open(*beatPath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error opening target Beat: %+v\n", err)
+		os.Exit(1)
+	}
+	esBeatsInfo, err := esBeats.Stat()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error getting file info of elastic/beat: %+v\n", err)
+		os.Exit(1)
+	}
+	beatInfo, err := beat.Stat()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error getting file info of target Beat: %+v\n", err)
 		os.Exit(1)
 	}
 
-	if len(beatFieldsPaths) == 0 {
-		fmt.Println("No field files to collect")
-		err = fields.AppendFromLibbeat(*esBeatsPath, *beatPath)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Cannot generate global fields.yml for %s: %+v\n", name, err)
-			os.Exit(2)
+	if len(beatFieldsPaths) == 0 && os.SameFile(esBeatsInfo, beatInfo) {
+		if !*toStdout {
+			fmt.Println("No field files to collect")
```

Comment:
You might consider using the stdlib `log` package in the file and the util packages. Just initialize it with `log.SetFlags()` to remove the timestamps. All messages will go to stderr all the time.

And all the places where it logs and exits can use a `log.Fatal*`.

Reply:
I would rather keep it as it is right now, because this is the "idiomatic" Beats way when printing to console during scripts or commands. I see that you started to use `log` in mage files, but apart from that `fmt` package is used all over the codebase.

Reply:
@andrewkroh Wasn't aware of this feature. We definitively should start using this in the future.